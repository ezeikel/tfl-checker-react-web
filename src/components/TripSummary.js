import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import formatMoney from "../utils/formatMoney";
import Mode from "./Mode";

const Wrapper = styled.section`
  flex: 0 0 auto;
  display: flex;
  align-items: center;
  width: 100%;
  padding: 32px;
  background-color: #fff;
  border-radius: 4px;
  .timing {
    display: flex;
    justify-content: space-between;
  }
`;

const Legs = styled.div`
  flex: 1 0 auto;
  display: flex;
  align-items: center;
  > div {
    & + div {
      margin-left: 16px;
    }
  }
`;

const TotalFare = styled.div`
  flex: 0 0 auto;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  color: #31cc71;
  margin-right: 32px;
`;

const Duration = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  span {
    &:first-of-type {
      font-size: 2.4rem;
      font-weight: bold;
    }
    &:last-of-type {
      font-size: 1.6rem;
    }
  }
`;

const TripSummary = ({ journey, className }) => {
  if (Object.keys(journey).length === 0) return null;

  return (
    <Wrapper className={className}>
      <Legs>
        {journey.legs.map((leg, i) => (
          <Mode
            key={i} // eslint-disable-line react/no-array-index-key
            leg={leg}
          />
        ))}
      </Legs>
      <TotalFare>
        {formatMoney(journey.fare ? journey.fare.totalCost : 0)}
      </TotalFare>
      <Duration>
        <span>{journey.duration}</span>
        <span>min</span>
      </Duration>
    </Wrapper>
  );
};

TripSummary.propTypes = {
  className: PropTypes.string.isRequired,
  journey: PropTypes.shape({
    fare: PropTypes.shape({
      totalCost: PropTypes.number,
    }),
    duration: PropTypes.number,
    legs: PropTypes.arrayOf(PropTypes.shape({})),
  }).isRequired,
};

export default TripSummary;
