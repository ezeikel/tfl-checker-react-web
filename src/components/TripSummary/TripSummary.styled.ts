import styled from "styled-components";

export const Wrapper = styled.section`
  flex: 0 0 auto;
  display: flex;
  align-items: center;
  width: 100%;
  padding: 32px;
  background-color: #fff;
  border-radius: 4px;
  .timing {
    display: flex;
    justify-content: space-between;
  }
`;

export const Legs = styled.div`
  flex: 1 0 auto;
  display: flex;
  align-items: center;
  > div {
    & + div {
      margin-left: 16px;
    }
  }
`;

export const TotalFare = styled.div`
  flex: 0 0 auto;
  display: flex;
  font-size: 16px;
  font-weight: bold;
  color: #31cc71;
  margin-right: 32px;
`;

export const Duration = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  span {
    &:first-of-type {
      font-size: 2.4rem;
      font-weight: bold;
    }
    &:last-of-type {
      font-size: 1.6rem;
    }
  }
`;
