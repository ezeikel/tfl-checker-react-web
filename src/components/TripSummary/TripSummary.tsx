import formatMoney from "../../utils/formatMoney";
import Mode from "../Mode/Mode";
import { Wrapper, Legs, TotalFare, Duration } from "./TripSummary.styled";
import { Journey } from "../../../types";

type TripSummaryProps = {
  className?: string;
  journey: Journey;
};

const TripSummary = ({ journey, className }: TripSummaryProps) => {
  if (Object.keys(journey).length === 0) return null;

  return (
    <Wrapper className={className}>
      <Legs>
        {journey.legs.map((leg, i) => (
          <Mode
            key={i} // eslint-disable-line react/no-array-index-key
            leg={leg}
          />
        ))}
      </Legs>
      <TotalFare>
        {formatMoney(journey.fare ? journey.fare.totalCost : 0)}
      </TotalFare>
      <Duration>
        <span>{journey.duration}</span>
        <span>min</span>
      </Duration>
    </Wrapper>
  );
};

export default TripSummary;
