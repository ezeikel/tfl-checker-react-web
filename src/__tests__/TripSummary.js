import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import TripSummary from '../components/TripSummary';

const middlewares = [thunk]; // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);

let mockProps;

beforeEach(() => {
  mockProps = {
    className: "",
    journey: {
      fare: {
        totalCost: 200
      },
      duration: 800,
      legs: []
    },
    onSetSelectedTrip: jest.fn(),
  };
});

test('renders without crashing', () => {
  const store = mockStore();

  const { container } = render(
    <TripSummary {...mockProps} store={store} /> // eslint-disable-line react/jsx-props-no-spreading
  );
  expect(container).toMatchSnapshot();
});
