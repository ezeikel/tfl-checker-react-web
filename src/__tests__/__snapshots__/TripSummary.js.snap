// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <section
    class="sc-AxirZ lfnEPT"
  >
    <div
      class="sc-AxiKw blrKOd"
    />
    <div
      class="sc-AxhCb jQYqSy"
    >
      Â£2.00
    </div>
    <div
      class="sc-AxhUy hEXbAE"
    >
      <span>
        800
      </span>
      <span>
        min
      </span>
    </div>
  </section>
</div>
`;
